src/control/server/faults.go

```diff
@@ -65,6 +67,26 @@ var (
 	)
 )
 
+func FaultPoolNvmeTooSmall(reqBytes uint64) *fault.Fault {
+	return serverFault(
+		code.ServerPoolNvmeTooSmall,
+		fmt.Sprintf("requested NVMe capacity (%s) for pool is too small",
+			humanize.Bytes(reqBytes)),
+		fmt.Sprintf("specify an NVMe capacity larger than %s when creating a pool",
+			minNvmeSize),
+	)
+}
+
+func FaultPoolScmTooSmall(reqBytes uint64) *fault.Fault {
+	return serverFault(
+		code.ServerPoolScmTooSmall,
+		fmt.Sprintf("requested SCM capacity (%s) for pool is too small",
+			humanize.Bytes(reqBytes)),
+		fmt.Sprintf("specify an SCM capacity larger than %s when creating a pool",
+			minScmSize),
+	)
+}
+
```

Comment:
It would be nice to future-proof these a bit. Currently, it's true that the values supplied to dmg are per-target, rather than per-pool. If we assume that we'll retain the ability to set per-target values in addition to a future capability of just setting per-pool values, then it probably makes sense to rename these to FaultTarget(Nvme|Scm)TooSmall and update the text accordingly. This may also help to reduce a bit of user confusion until we add that per-pool capability.

Reply:
The values supplied to dmg are per io server not per target and not the total pool size. That's why we have to multiply the minimum target allocation by the number of targets to get a viable minimum value to compare against https://github.com/daos-stack/daos/blob/master/src/mgmt/srv_target.c#L488
According to @johannlombardi there might be a future change where instead the parameter is treated as total pool size. with this in mind I think the naming should stay as it is if that's okay.